### DISCUSSION TITLE
How to execute the custom query of same module in the query handler of the same module,please?

### DISCUSSION THREAD

**Discussion by effapp** [kgrzybek/modular-monolith-with-ddd#156] [Q&A]
How to execute another custom query of same module in the query handler of the same module,please?

example:

I created a custom query handler that named "CustomQueryHandler" in UserAccessModule, how to execute query "GetUserQuery" of UserAccessModule in handle of CustomQueryHandler of UserAccessModule?

**Comment by kgrzybek** [comment]
Hi @effapp 

In general, I don't execute another query from the query handler (the same for a command handling). I think this level of indirection is not needed. 

If I want to reuse some logic, I create a separate class/method for this behavior and reference it directly.
